### PR TITLE
feat/authentication/cleanup : Clean up code from M1 : `ui.authentication` package

### PR DESCRIPTION
**Pull request by charliemangano**
# Clean up code from M1 : `ui.authentication` package

## Description
This PR introduces code cleanup and refactoring, as well as new documentation for the files in package `ui.authentication`, including corresponding tests. It closes issue #124 and part of #84.

## Changes
**AuthComponents.kt:**  
Renamed:  
* `AuthComponents.kt` to `AuthenticationComponents.kt`;
* `Auth...` components to `Authentication...`;
* `AuthButton` to `AuthenticationSubmitButton` for clarity.

Deleted redundant `AuthInstruction` and `AuthSecondInstruction` as they don't actually modularize the code.
Moved error message check to `AuthEmailInput` and `AuthPasswordInput` instead of having them outside.
Modified signatures to have optional instead of mandatory parameters.
Added documentation

**SignIn.kt and SignUp.kt:**
Extracted relevant values to constant variables. Extracted `onClick` for submit button to auxiliary method.
Modified email and password checking methods to return a `Boolean`.
Removed redundant or optional arguments in calls to `AuthenticationComponents.kt` componenents.

#### Added
None
#### Modified
- `app/src/main/java/com/android/periodpals/ui/authentication/SignIn.kt`, `app/src/main/java/com/android/periodpals/ui/authentication/SignUp.kt`: code cleanup and documentation
- `app/src/main/java/com/android/periodpals/ui/components/AuthComponents.kt`: renamed to `AuthenticationComponents.kt`, code cleanup and documentation
- `app/src/main/java/com/android/periodpals/resources/C.kt`: refactored authentication test tags
- `app/src/androidTest/java/com/android/periodpals/ui/authentication/SignInTest.kt`, `app/src/androidTest/java/com/android/periodpals/ui/authentication/SignUpTest.kt`: propagated changes to test tags
#### Removed
None
## Dependencies Added
None

## Testing
All tests still pass.

### CODE REVIEW

File: app/src/main/java/com/android/periodpals/ui/authentication/SignIn.kt [hunk 1]
```diff
@@ -245,3 +212,89 @@ fun GoogleButton(onClick: () -> Unit, modifier: Modifier = Modifier, testTag: St
...
+ * @param setEmailErrorMessage A function to set the error message for the email field.
+ * @param password The password entered by the user.
+ * @param setPasswordErrorMessage A function to set the error message for the password field.
+ * @param authenticationViewModel The ViewModel that handles authentication logic.
+ * @param userState The current state of the user authentication.
+ * @param context The context used to show Toast messages.
+ * @param navigationActions The navigation actions to navigate between screens.
+ * @return A lambda function to be called on button click.
+ */
+private fun attemptSignIn(
```
Comment (lazarinibruno) [inline app/src/main/java/com/android/periodpals/ui/authentication/SignIn.kt:229]: I find that the code inside this function is hard to read, given the multiple nested if statements.

Comment (charliemangano) [reply app/src/main/java/com/android/periodpals/ui/authentication/SignIn.kt:229]: I'll try and take care of that, thanks for pointing it out! Same remark for `attemptSignUp` in `SignUp.kt` or is that one okay?

**Review by lazarinibruno — CHANGES_REQUESTED** [review]
## Summary

Thanks for your contribution!

- Very good job on decreasing the duplication of code.
  
- Improves mainteinance by extracting constant values.
  

## Improvements

In `SignIn.kt` I would refactor the `attemptSignIn` function to something like:

```kotlin
if (!isEmailValid(email, setEmailErrorMessage) || 
    !isPasswordValid(password, setPasswordErrorMessage) {
    return { Toast.makeText(context, INVALID_ATTEMPT_TOAST, Toast.LENGTH_SHORT).show() }
}

// Attempt log in
authenticationViewModel.logInWithEmail(email, password)
authenticationViewModel.isUserLoggedIn()


return {
    val loginSuccess = userState is UserAuthState.Success
    if (loginSuccess) {
        Toast.makeText(context, SUCCESSFUL_SIGN_IN_TOAST, Toast.LENGTH_SHORT).show()
        navigationActions.navigateTo(Screen.PROFILE)
    } else {
        Toast.makeText(context, FAILED_SIGN_IN_TOAST, Toast.LENGTH_SHORT).show()
    }
}
```

## Steps before merge

Just the modification I suggested. Otherwise, code LGTM.

**Review by agonzalez-r — APPROVED** [review]
Code is consistent and much more clean. 
Thank you for this hard work! 

LGTM

**Review by lazarinibruno — APPROVED** [review]
Code is much readable now!

**Pull request merged**